custom_components/iec/coordinator.py

```diff
@@ -629,11 +629,18 @@ async def _estimate_bill(self, contract_id, device_number, is_private_producer,
         if not is_private_producer:
             try:
                 devices_by_id: Devices = await self._get_devices_by_device_id(device_number)
-                last_meter_read = int(devices_by_id.counter_devices[0].last_mr)
-                last_meter_read_date = devices_by_id.counter_devices[0].last_mr_date
-                phase_count = devices_by_id.counter_devices[0].connection_size.phase
-                connection_size = (devices_by_id.counter_devices[0].
-                                    connection_size.representative_connection_size)
+
+                counter_devices = devices_by_id.counter_devices
+
+                if counter_devices is not None:
+                    first_counter_device = counter_devices[0]
+                    connection_size = first_counter_device.connection_size
```

Comment:
To be honest, I already had this in local branch.

```suggestion
                if devices_by_id.counter_devices and len(devices_by_id.counter_devices) >= 1:
                    last_meter_read = int(devices_by_id.counter_devices[0].last_mr)
                    last_meter_read_date = devices_by_id.counter_devices[0].last_mr_date
                    phase_count = devices_by_id.counter_devices[0].connection_size.phase
                    connection_size = (devices_by_id.counter_devices[0].
                                        connection_size.representative_connection_size)
                else:
                    _LOGGER.warning("Failed to get Last Device Meter Reading, trying another way...")
```

Reply:
Thanks!